pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -128,13 +155,34 @@ public static PropertiesConfiguration fromFile(File file, boolean setIOFactory,
     return config;
   }
 
+  /**
+   * save the segment metadata configuration content into the provided file based on the version header.
+   * @param propertiesConfiguration a {@link PropertiesConfiguration} instance.
+   * @param file a {@link File} instance.
+   * @param versionHeader a {@link String} instance.
+   */
+  public static void saveSegmentMetadataToFile(PropertiesConfiguration propertiesConfiguration, File file,
```

Comment:
mark `@Nullable versionHeader`

Reply:
not merging two checks because, in-future, we might want to set the version header with a different IO factory.